Python/ceval.c

```diff
@@ -2766,29 +2740,24 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             }
             PyObject *type, *value, *traceback;
             _PyErr_StackItem *exc_info;
-            type = POP();
             value = POP();
-            traceback = POP();
-            ASSERT_EXC_TYPE_IS_REDUNDANT(type, value);
+            assert(value);
+            assert(PyExceptionInstance_Check(value));
+            type = Py_NewRef(PyExceptionInstance_Class(value));
+            traceback = PyException_GetTraceback(value);
             Py_DECREF(POP()); /* lasti */
             _PyErr_Restore(tstate, type, value, traceback);
```

Comment:
OOI, do you plan to add a version of `_PyErr_Restore` that just takes `tstate` and `value` as arguments?

Reply:
Yes